golang/cosmos/types/address_hooks.go

```diff
@@ -176,7 +174,7 @@ func extractBaseTransferData(transferData transfertypes.FungibleTokenPacketData,
 // address is exactly the original address.
 // If newPacket is not nil, it is populated with a new transfer packet whose
 // corresponding Sender or Receiver is replaced with the extracted base address.
-func ExtractBaseAddressFromPacket(cdc codec.Codec, packet ibcexported.PacketI, role AddressRole, newPacket *channeltypes.Packet) (string, error) {
+func ExtractBaseAddressFromPacket(cdc codec.Codec, packet ibcexported.PacketI, role AddressRole, newPacket ibcexported.PacketI) (string, error) {
```

Comment:
Why not require that `newPacket` be an `IBCPacket`?

Reply:
> Why not require... an `IBCPacket`?

I agree with this suggestion for `address_hooks.go`.  It's a more principled approach than sniffing what type of `PacketI` the `newPacket` argument is.  We can leave it up to the caller to copy from an `IBCPacket` to an ibc-go channel `Packet` when necessary.